src/app/cli/src/cli_value_parser.rs

```diff
@@ -74,6 +74,18 @@ pub(crate) fn dataset_ref_remote(s: &str) -> Result<odf::DatasetRefRemote, Strin
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
+pub(crate) fn transfer_dataset_ref_remote(s: &str) -> Result<odf::TransferDatasetRef, String> {
+    match odf::TransferDatasetRef::from_str(s) {
+        Ok(push_dataset_ref) => Ok(push_dataset_ref),
+        Err(_) => Err("Remote reference should be in form: `did:odf:...` or \
```

Comment:
`TransferDatasetRef` is used in `--to` argument, but does `did:odf:...` make sense in this position?

From our conversation I though in `--to` we will only allow:
- `Url`
- `RepoName`
- `RepoName/DatasetName` (defaults to current account)
- `RepoName/AccountName/DatasetName`

Also perhaps `DatasetPushTarget` is a better name for this type.

Reply:
Yeah, you are right! I forgot that it will pass ID etc refs. Changed